src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -1728,9 +1733,9 @@ namespace winrt::TerminalApp::implementation
         hostingTab.TaskbarProgressChanged({ get_weak(), &TerminalPage::_SetTaskbarProgressHandler });
     }
 
-    void TerminalPage::_RegisterPaneEvents(std::shared_ptr<Pane>& pane)
+    void TerminalPage::_RegisterPaneEvents(const TerminalApp::TerminalPaneContent& paneContent)
```

Comment:
more of a `RegisterPaneContentEvents`

Reply:
wait a second, it's not even that. It's `RegisterBigTimeEncapsulationViolatingTerminalPaneContentEvents`

Reply:
brainstorming

okay what if TerminalTab listened for this coming off a pane, then bubbled the event up with the TerminalPaneContent as the sender. So it'd be _tab.RestartPaneRequested(TerminalPaneContent, object) instead

that would be 5% less encapsulation violating